src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5406,13 +5487,89 @@ private void BindMemberAccessReportError(BoundMethodGroup node, DiagnosticBag di
                 {
                     Error(diagnostics, ErrorCode.ERR_NoSuchMemberOrExtensionNeedUsing, name, boundLeft.Type, plainName, "System");
                 }
+                else if (boundLeft.Type.IsTupleType && IsMissingMemberAnInferredTupleName(boundLeft, plainName))
+                {
+                    Error(diagnostics, ErrorCode.ERR_TupleInferredNamesNotAvailable, name, boundLeft.Type, plainName,
+                        new CSharpRequiredLanguageVersion(LanguageVersion.CSharp7_1));
+                }
                 else
                 {
                     Error(diagnostics, ErrorCode.ERR_NoSuchMemberOrExtension, name, boundLeft.Type, plainName);
                 }
             }
         }
 
+        /// <summary>
+        /// When reporting a missing member error on a tuple type, determine whether that member
+        /// could be an element with an inferred name (had the language version been above 7.1).
+        /// </summary>
+        private bool IsMissingMemberAnInferredTupleName(BoundExpression boundLeft, string plainName)
+        {
+            Debug.Assert(boundLeft.Type.IsTupleType);
+            if (Compilation.LanguageVersion >= LanguageVersion.CSharp7_1)
```

Comment:
Given the name of the method, it feels like the `LanguageVersion` check should be handled outside. And then this method could be `static`.